### PR TITLE
Alexmont/dpe work

### PR DESCRIPTION
**Pull request by alexmontAmazon**
## What does this PR do?
- fixes changeNotify only inheriting one level of callbacks from its ancestry
- adds ability to invoke multiple function Dom::Values from a Dom::Array
- implements Dom::Array function handling for multi-edit in the AggregateAdapter

_Please add links to any issues, RFCs or other items that are relevant to this PR._
Fixes: https://github.com/o3de/o3de/issues/15727

## How was this PR tested?
 locally with the Editor

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp [hunk 1]
```diff
@@ -428,14 +428,52 @@ namespace AZ::Reflection
...
+                    {
+                        if (!existingValue->IsArray())
+                        {
+                            Dom::Value arrayValue;
+                            arrayValue.SetArray();
+                            arrayValue.ArrayPushBack(*existingValue);
+                            *existingValue = arrayValue;
+                        }
+                        // existing value is an array, add parents changeNotify to it
+                        addValueToArray(*parentValue, *existingValue);
```
Comment (michabr) [inline Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:468]: Do we need to update cached attributes here?

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp [hunk 2]
```diff
@@ -428,14 +428,52 @@ namespace AZ::Reflection
                 // Cache attributes for the current node. Attribute data will be used in ReflectionAdapter.
                 CacheAttributes();
 
-                // Inherit the change notify attribute from our parent
-                const Name changeNotify = Name("ChangeNotify");
-                if (auto changeNotifyValue = Find(changeNotify); !changeNotifyValue || changeNotifyValue->IsNull())
+                // Inherit the change notify attribute from our parent, if it exists
+                const auto changeNotifyName = DocumentPropertyEditor::Nodes::PropertyEditor::ChangeNotify.GetName();
+                auto parentValue = Find(Name(), changeNotifyName, parentData);
```
Comment (michabr) [inline Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:433]: Will this find the first ancestor with a value, or just the immediate parent?

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp [hunk 3]
```diff
@@ -428,14 +428,52 @@ namespace AZ::Reflection
...
-                    if (changeNotifyValue && !changeNotifyValue->IsNull())
+                    Dom::Value* existingValue = nullptr;
+                    for (auto it = nodeData->m_cachedAttributes.begin(); !existingValue && it != nodeData->m_cachedAttributes.end(); ++it)
                     {
-                        nodeData->m_cachedAttributes.push_back({ Name(), changeNotify, *changeNotifyValue });
+                        if (it->m_name == changeNotifyName)
+                        {
+                            existingValue = &it->m_value;
+                        }
+                    }
```
Comment (forkercat) [inline Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:443]: minor: Can use `std::find_if` here

```cpp
auto result = std::find_if(m_cachedAttributes.begin(), m_cachedAttributes.end(), 
    [](const auto& value) { return value.m_name == changeNotifyName; });
if (result != m_cachedAttributes.end())
{
    AZ::Dom::Value& existingValue = *result; // or ptr
    ...
}
```

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp (hunk 2 above)
Comment (forkercat) [reply Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:433]: I think it is just the immediate parent. But at the time we're processing the here node, the parent node should have already been processed. So the current node should also inherit all values from its ancestry I guess.

**Review by forkercat — APPROVED** [review]
Looks good :)

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp (hunk 2 above)
Comment (michabr) [reply Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:433]: That makes sense, thanks.

**Review by michabr — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp (hunk 2 above)
Comment (alexmontAmazon) [reply Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:433]: exactly, we all only inherit from our parents, but that doesn't mean that I don't have my great grandfather's right elbow, which everyone knows I totally do.

**Review by srikappa-amzn — APPROVED** [review]

**Review by dtamkin1 — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp (hunk 3 above)
Comment (alexmontAmazon) [reply Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp:443]: done :)

**Pull request merged**